src/compiler/checker.ts

```diff
@@ -8744,7 +8758,6 @@ namespace ts {
                 let type = signature.target ? instantiateType(getReturnTypeOfSignature(signature.target), signature.mapper!) :
                     signature.unionSignatures ? getUnionType(map(signature.unionSignatures, getReturnTypeOfSignature), UnionReduction.Subtype) :
                     getReturnTypeFromAnnotation(signature.declaration!) ||
-                    isJSConstructor(signature.declaration) && getJSClassType(getSymbolOfNode(signature.declaration!)) ||

```

Comment:
this special case also moves into the `Class|Function` case of resolveAnonymousTypeMembers, where the correct answer is now just `classType`.

Note that call signatures of constructor functions now correctly fall through to `getReturnTypeFromBody`, where before they *also* claimed to return the class type from `getJSClassType`.